src/System.ComponentModel.EventBasedAsync/tests/BackgroundWorkerTests.cs

```diff
@@ -148,45 +148,96 @@ private void DoWorkExpectCancel(object sender, DoWorkEventArgs e)
             manualResetEvent3.Set();
         }
 
+        #endregion
+
         [Fact]
         public void TestThrowExceptionInDoWork()
         {
-            var bw = new TestWorker();
-            bw.DoWork += DoWorkWithException;
-            bw.RunWorkerAsync("Exception");
+            const string expectedArgument = "Exception";
+            const string expectedExceptionMsg = "Exception from DoWork";
+            var bw = new BackgroundWorker();
+
+            bw.DoWork += (sender, e) =>
+            {
+                Assert.Same(bw, sender);
+                Assert.Same(expectedArgument, e.Argument);
+                throw new TestException(expectedExceptionMsg);
+            };
+
+            bw.RunWorkerCompleted += (sender, e) =>
+            {
+                TestException ex = Assert.Throws<TestException>(() => e.Result);
+                Assert.Equal(expectedExceptionMsg, ex.Message);
+            };
+
+            bw.RunWorkerAsync(expectedArgument);
+        }
+
+        [Fact]
+        public void CtorTest()
+        {
+            var bw = new BackgroundWorker();
+            Assert.False(bw.IsBusy);
+            Assert.False(bw.WorkerReportsProgress);
+            Assert.False(bw.WorkerSupportsCancellation);
+            Assert.False(bw.CancellationPending);
         }
 
-        private void DoWorkWithException(object sender, DoWorkEventArgs e)
+        [Fact]
+        public void RunWorkerAsyncTwice()
+        {
+            var bw = new BackgroundWorker();
+            bw.DoWork += (sender, e) => Wait(3 * TimeoutShort);
+            bw.RunWorkerAsync();
+            Assert.Throws<InvalidOperationException>(() => bw.RunWorkerAsync());
+        }
+
+        [Fact]
+        public void TestCancelAsyncWithoutCancellationSupport()
         {
-            Assert.Equal("Exception", e.Argument);
-            throw new TestException("Exception from DoWork");
+            var bw = new BackgroundWorker() { WorkerSupportsCancellation = false };
+            bw.DoWork += (sender, e) => Wait(3 * TimeoutShort);
+            Assert.Throws<InvalidOperationException>(() => bw.CancelAsync());
```

Comment:
See previous comment; this test has a race condition due to the timeout, and we should instead use synchronization to ensure we get the interleaving we want.